feat(config): add docker cli config (#22684)
renovatebot/renovate: docs/usage/self-hosted-configuration.md

!!! note
    Dangling containers are only removed when Renovate runs again with the same prefix.

## dockerCliOptions

> Adds command-line docker CLI options to Renovate sidecar Docker containers.

Reviewer: ```suggestion
You can use `dockerCliOptions` to pass Docker CLI options to Renovate's sidecar Docker containers.
```

> For example, `--memory=1g --cpus="1.5"`.

Reviewer: I'd like a real config option example here, something similar to:

```json
{
  "dockerCliOptions": ["this-is-a-bad-config-do-not-copy"]
}
```

Reviewer 2: It's not completely bad or good

Reviewer 3: 🤔  think those are already valid options.

> Read the [Docker runtime resource constraints](https://docs.docker.com/config/containers/resource_constraints/) for more information.

Reviewer: It feels like we're missing an explanation on our side here. How about a structure like this:

```markdown
One line summary of config option.

Explanation about _why_ and _when_ a user may need this config option.
Expand on the `--memory=1g --cpus="1,5"` stuff.

Real config example in proper JSON.

To learn more about ... read the Docker runtime resource constrains.
```

Author: updated

!!! note
    Dangling containers are only removed when Renovate runs again with the same prefix.

## dockerCliOptions

> Adds command-line docker cli options to Renovate sidecar Docker containers.

Reviewer 2: ```suggestion
Adds command-line docker CLI options to Renovate sidecar Docker containers.
```

> for example `--memory=1g --cpus="1.5"`

Reviewer 2: ```suggestion
For example, `--memory=1g --cpus="1.5"`.
```

Read the [Docker runtime resource constraints](https://docs.docker.com/config/containers/resource_constraints/) for more information.

> Note: there are other config options available for docker side-car container that adds cli options as well for running and also cleaning containers from previous runs
> 

Reviewer 2: ```suggestion
```

!!! note
    Dangling containers are only removed when Renovate runs again with the same prefix.

## dockerCliOptions

You can use `dockerCliOptions` to pass Docker CLI options to Renovate's sidecar Docker containers.

> For example, `{"dockerCliOptions": "--memory=4g"}` will add a cli flag to the docker run command that will cap the maximum amount of memory Renovate's sidecar Docker container can use to 4 gigabytes.

Reviewer: ```suggestion
For example, `{"dockerCliOptions": "--memory=4g"}` will add a CLI flag to the `docker run` command that limits the amount of memory Renovate's sidecar Docker container can use to 4 gigabytes.
```

> Click here to learn more about [Docker runtime resource constraints](https://docs.docker.com/config/containers/resource_constraints/).

Reviewer: ```suggestion
Read the [Docker Docs, configure runtime resource contraints](https://docs.docker.com/config/containers/resource_constraints/) to learn more.
```
